### ISSUE TITLE
[BUG]: restrict_to_aoi not implemented

### ISSUE THREAD

**Issue by rbavery** [CosmiQ/solaris#327]
`restrict_by_aoi` is defined as a boolean argument in the raster tiler but doesn't appear to be implemented. I was thinking of implementing this but instead of making it a boolean, the portion of the raster that falls within the aoi would remain and the portion of the tile that falls outside the aoi would be filled with the mean of the training dataset. I think this type of filling would work since labels are not present outside of the aoi_boundary (by definition of how `aoi_boundary` should be used). Does this sound good @nrweir ?

**Comment by nrweir** [comment]
Hey @rbavery,

That sounds great! My long-term vision for this is to have a few different options - including reflecting values across the boundary (similar to what's often implemented for image scaling algorithms for deep learning) and a couple of others. Maybe we can start with two options:

1. Use the mean;
2. Use a custom user-provided value (e.g. so they can use `0` or something if they really want to)

...while leaving flexibility to add additional options in the future. What do you think?

**Comment by rbavery** [comment]
Sounds good to me! The easiest thing to do would be to use the mean of the image used for tiling. But we could also use the mean of all images that will be tiled by supplying a directory. The former is simpler but the latter means a consistent fill value will be used across all tiles in cases where multiple images are tiled, which would make visualization of the tiles easier. I'm leaning toward the latter option as the default, what do you think @nrweir ?

**Comment by nrweir** [comment]
Ideally, we enable both: allow users to specify a fill value (which they could derive from the full-dataset mean) or solaris could derive the mean within the image itself. Would essentially need to provide an argument specifying either a value to fill with, or allow users to specifiy `'mean'`, in which case solaris will get the image's mean value and use that. e.g. a new argument `nodata_fill=None` where the user can provide, for example, `nodata_fill=50` (or whatever other specific value they want) or `nodata_fill='mean'` in which case solaris will get it from the image for them.

**Comment by yangsiyu007** [comment]
I just wanted to say filling in with custom values (0 in particular) would be great to have as one of the options at least. Computing the mean can be time-consuming over a large area, and being able to mask the label raster with 0 where the imagery tile has no data would be useful (i.e. to not have these contribute to the loss during training).

**Comment by rbavery** [comment]
This is solved by https://github.com/CosmiQ/solaris/pull/331 @yangsiyu007 . there's now an option to fill with either the mean of the source image or a chosen `fill_value`. See the `fill_all_nodata` method of the `RasterTiler` class: https://github.com/CosmiQ/solaris/blob/dev/solaris/tile/raster_tile.py#L455

And https://github.com/CosmiQ/solaris/blob/dev/solaris/vector/mask.py#L978

**Comment by yangsiyu007** [comment]
Good job @rbavery ! Looking forward to trying it out
